### ISSUE TITLE
Parsing Issues

### ISSUE THREAD

**Issue by Philipp-Sc** [mliebelt/pgn-parser#90]
Hi Markus,

thank you for all the work you put into this. 
I am using the pgn-parser for my <a href="https://github.com/Philipp-Sc/learning"> project</a>, to compute meta data from a pgn database.

I ran into the following issue:

Provided the pgn contains the following string as part of the game.

`1. d4 {[%depth20 +0.35] [%depth1 +0.34]} 1. ... Nf6 {[%depth20 +0.24] [%depth1 +0.13]} 2. 
Nf3 {[%depth20 +0.26] [%depth1 +0.25]} 2. ... c5 {[%depth20 +0.82] [%depth1 +0.59]} 3. e3`

the parsing will fail.  After debugging this the reason seems to be the "1. ...", "2. ..." etc syntax. After removing all the redundant information I end up with:

`1. d4 {[%depth20 +0.35] [%depth1 +0.34]} Nf6 {[%depth20 +0.24] [%depth1 +0.13]} 2. 
Nf3 {[%depth20 +0.26] [%depth1 +0.25]} c5 {[%depth20 +0.82] [%depth1 +0.59]} 3. e3`

the parsing is succesful but the comments are not included in commentAfter.
To fix this removing all strings "%depth" fixed this issue.

I am not sure what happens here. I imagine this could be a security issue, where one introduces code into a pgn that gets executed. 

The code used to parse the pgn is:
`       const response = await fetch('/games/'+types[type]);`

`       const text = await response.text();  `

`       var pgn = parser.parse(text, {startRule: "games"})  `

Best regards,
Philipp

**Comment by mliebelt** [comment]
The reason is the grammar used. Here is the relevant part of the rule:

```
innerComment
...
  / ws bl "%" ac:stringNoQuot wsp nbr+ br ws tail:(ic:innerComment { return ic })*
      { return tail[0] }
``
This returns just the tail (which is empty in your case, so the action comments are ignored. See the [relevant part of the spec](https://github.com/mliebelt/pgn-spec-commented/blob/main/pgn-spec-supplement.md#3-proposalshttps://github.com/mliebelt/pgn-spec-commented/blob/main/pgn-spec-supplement.md#3-proposals). As the supplemental "specification" (never official) does not list your action comments, they are not included.

Do you have a list of possible action comments referenced somewhere? 

It would be possible to just include a generic rule here. I will give it a try, and publish a new version then.

Thank you for asking that, good idea!

**Comment by mliebelt** [comment]
The relevant [section about the move numbers](https://github.com/mliebelt/pgn-spec-commented/blob/main/pgn-specification.md#8221-import-format-move-number-indications) is short and crisp:

"PGN import format move number indications may have zero or more period characters following the digit sequence that gives the move number; one or more white space characters may appear between the digit sequence and the period(s)."

And this is how it is implemented. It could be more sloppy, but specs are the way to go (in my opinion).

**Comment by mliebelt** [comment]
Published version 1.3.5 to NPM, that should include the change. Hope that fixes your problem with the missing action comments.

**Comment by Philipp-Sc** [comment]
I agree with the move numbers, the syntax uses less space and therefore is better. I did not know the pgn did not have the official syntax, it was a little annoying to figure out that the numbering caused the errors.

Thank you for adding the arbitrary action commands, that will make it easier for me. Hopefully for others as well.
